Source/BugsnagNotifier.m

```diff
@@ -218,12 +218,23 @@ - (id)initWithConfiguration:(BugsnagConfiguration *)initConfiguration {
     if ((self = [super init])) {
         self.configuration = initConfiguration;
         self.state = [[BugsnagMetaData alloc] init];
+        NSString *notifierName =
+#if TARGET_OS_TV
+            @"tvOS Bugsnag Notifier";
+#elif TARGET_IPHONE_SIMULATOR || TARGET_OS_IPHONE
+            @"iOS Bugsnag Notifier";
+#elif TARGET_OS_MAC
+            @"OSX Bugsnag Notifier";
+#else
+            @"Bugsnag Objective-C";
+#endif
         self.details = [@{
-            BSGKeyName : @"Bugsnag Objective-C",
+            BSGKeyName : notifierName,
```

Comment:
Previously, the name was set in `start`, this change moves it into the initializer so it can be overwritten if needed and the changes aren't lost when start is called.